examples/mosquitto/README.md

```diff
@@ -0,0 +1,225 @@
+best use the package that comes with the distribution. On Ubuntu this is done by
+running:
+```bash
+sudo apt install mosquitto
+```
+
+Once installed, make sure to stop it as we will be running it with the spiffe
+helper instead:
+```bash
+sudo systemctl stop mosquitto
+```
+
+### 6. Register Mosquitto Broker Workload with Spire Server
+
+On `node0` run the following command:
+```bash
+cd /opt/spire
+./bin/spire-server entry create \
+                   -spiffeID spiffe://example.org/mosquitto-server \
+                   -parentID spiffe://example.org/node1 \
+                   -selector unix:user:mosquitto \
+                   -ttl 600 \
+                   -dns node1
+```
+
+For `-dns` we use the hostname that the client will use to connect to the
+broker. In this case we are using the hostname of `node1`. We also use the
+unix attestation selector with user `mosquitto` to identify the workload.
+
+For testing purposes we also set the `-ttl` to 10 minutes.
+
+### 6. Configure Spiffe Helper and Run Mosquitto Broker on Node1
+
+Download and build the spiffe helper on `node1`. Note that we will be running
+the spiffe helper with user `mosquitto` so make sure that the location where
+you install the spiffe-helper binary has the right permissions for reading the
+configuration files:
+```bash
+chmod -R o+r ./examples/mosquitto
+```
+
+Now start up the mosquitto broker using the spiffe helper:
+```bash
+sudo mkdir -p /opt/spire/certs/mosquitto
+sudo chown mosquitto:mosquitto /opt/spire/certs/mosquitto
+sudo -u mosquitto ./bin/spiffe-helper -config ./examples/mosquitto/helper.conf
+```
+
+### 7. Install Mosquitto Client on Node2
+
+Make sure to install the mosquitto client tools on node2. On Ubuntu this is done
+with:
+```bash
+sudo apt install mosquitto-clients
+```
+
+We will be running the client under a dedicated user called `mosquitto-client`.
+Create the user with:
+```bash
+sudo useradd mosquitto-client
+```
+
+### 7. Register the Mosquitto Client Workload with Spire Server
+
+On `node0` run the following command:
+```bash
+cd /opt/spire
+./bin/spire-server entry create \
+                   -spiffeID spiffe://example.org/mosquitto-client \
+                   -parentID spiffe://example.org/node2 \
+                   -selector unix:user:mosquitto-client \
+                   -ttl 600 \
+                   -dns jdoe
+```
+
+For `-dns` we use the username that the client will use to connect to the
+broker. In this case we are using the username `jdoe`. We also use the unix
+attestation selector with user `mosquitto-client` to identify the workload.
+
+For testing purposes we also set the `-ttl` to 10 minutes.
+
+### 8. Run the Mosquitto Client
+
+First create a directory for the client certificates:
+```bash
+mkdir -p /tmp/mosquitto/svids
+sudo chown mosquitto-client:mosquitto-client /tmp/mosquitto/svids
+```
+
+Copy over the script [./connect.sh] to `node2` and run it with:
```

Comment:
For the connect.sh you just reference the current directory but above you use `./examples/moquitto/..`. Just want to make sure we are consistent with the what directory we are running these commands from

Reply:
Good point! I will double check the references to the files.